packages/framework/src/Resources/elasticsearch/2.json

```diff
@@ -0,0 +1,151 @@
+          "tokenizer": "keep_special_chars",
+          "filter": [
+            "lowercase",
+            "asciifolding"
+          ]
+        },
+        "stemming": {
+          "tokenizer": "standard",
+          "filter": [
+            "lowercase",
+            "czech_stemmer",
+            "czech_stop",
+            "asciifolding"
+          ]
+        },
+        "edge_ngram_with_diacritic": {
+          "tokenizer": "keep_special_chars",
+          "filter": [
+            "edge_ngram",
+            "lowercase"
+          ]
+        },
+        "edge_ngram_without_diacritic": {
+          "tokenizer": "keep_special_chars",
+          "filter": [
+            "edge_ngram",
+            "lowercase",
+            "asciifolding"
+          ]
+        },
+        "edge_ngram_without_diacritic_html": {
+          "char_filter": "html_strip",
+          "tokenizer": "keep_special_chars",
+          "filter": [
+            "edge_ngram",
+            "lowercase",
+            "asciifolding"
+          ]
+        },
+        "edge_ngram_unanalyzed": {
+          "tokenizer": "keyword",
+          "filter": [
+            "edge_ngram"
+          ]
+        }
+      }
+    }
+  },
+  "mappings": {
+    "_doc": {
+      "properties": {
+        "name": {
+          "type": "text",
+          "analyzer": "stemming",
+          "fields": {
+            "full_with_diacritic": {
+              "type": "text",
+              "analyzer": "full_with_diacritic"
+            },
+            "full_without_diacritic": {
+              "type": "text",
+              "analyzer": "full_without_diacritic"
+            },
+            "edge_ngram_with_diacritic": {
+              "type": "text",
+              "analyzer": "edge_ngram_with_diacritic"
+            },
+            "edge_ngram_without_diacritic": {
+              "type": "text",
+              "analyzer": "edge_ngram_without_diacritic"
+            }
+          }
+        },
+        "catnum": {
+          "type": "text",
+          "analyzer": "keyword",
+          "fields": {
+            "edge_ngram": {
+              "type": "text",
+              "analyzer": "edge_ngram_unanalyzed"
+            }
+          }
+        },
+        "partno": {
+          "type": "text",
+          "analyzer": "keyword",
+          "fields": {
+            "edge_ngram": {
+              "type": "text",
+              "analyzer": "edge_ngram_unanalyzed"
+            }
+          }
+        },
+        "ean": {
+          "type": "text",
+          "analyzer": "keyword",
+          "fields": {
+            "edge_ngram": {
+              "type": "text",
+              "analyzer": "edge_ngram_unanalyzed"
+            }
+          }
+        },
+        "short_description": {
+          "type": "text",
+          "analyzer": "edge_ngram_without_diacritic"
+        },
+        "description": {
+          "type": "text",
+          "analyzer": "edge_ngram_without_diacritic_html"
+        }
+      }
+    }
+  }
+}
```

Comment:
Thanks! I have to tell this, because I'm proud about this idea: The White's implementation looked a bit confusing for us, so we decided for a different approach... Try to compare https://git.shopsys-framework.com/ssfw/ab-com.cz/blob/master/src/Shopsys/ShopBundle/Model/Product/Elasticsearch/ElasticsearchProductIndexConfig.php

Reply:
I like more your form for sure :)